src/basic/cpu-set-util.c

```diff
@@ -114,48 +119,19 @@ int parse_cpu_set_and_warn(
         return (int) ncpus;
 }
 
-int parse_cpu_set(
+int parse_cpu_set_and_warn(
                 const char *rvalue,
-                cpu_set_t **cpu_set) {
-
-        _cleanup_cpu_free_ cpu_set_t *c = NULL;
-        unsigned ncpus = 0;
-
-        assert(rvalue);
-
-        for (;;) {
-                _cleanup_free_ char *word = NULL;
-                unsigned cpu, cpu_lower, cpu_upper;
-                int r;
-
-                r = extract_first_word(&rvalue, &word, WHITESPACE ",", EXTRACT_QUOTES);
-                if (r == -ENOMEM)
-                        return r;
-                if (r <= 0)
-                        break;
-
-                if (!c) {
-                        c = cpu_set_malloc(&ncpus);
-                        if (!c)
-                                return -ENOMEM;
-                }
-
-                r = parse_range(word, &cpu_lower, &cpu_upper);
-                if (r < 0)
-                        return r;
-                if (cpu_lower >= ncpus || cpu_upper >= ncpus)
-                        return -EINVAL;
+                cpu_set_t **cpu_set,
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *lvalue) {
 
-                if (cpu_lower <= cpu_upper)
-                        for (cpu = cpu_lower; cpu <= cpu_upper; cpu++)
-                                CPU_SET_S(cpu, CPU_ALLOC_SIZE(ncpus), c);
-        }
+        assert(lvalue);
 
-        /* On success, sets *cpu_set and returns ncpus for the system. */
-        if (c) {
-                *cpu_set = c;
-                c = NULL;
-        }
+        return parse_cpu_set_internal(rvalue, cpu_set, true, unit, filename, line, lvalue);
+}
 
-        return (int) ncpus;
+int parse_cpu_set(const char *rvalue, cpu_set_t **cpu_set) {
+        return parse_cpu_set_internal(rvalue, cpu_set, false, NULL, NULL, 0, NULL);
```

Comment:
Not that it matters much, but those two wrapper functions could be made `static inline` in the `.h` file to reduce the line count and number of symbols a bit.

Reply:
OK. Updated.